### PR TITLE
GRS - Send Customers to GRS

### PR DESCRIPTION
**Pull request by dm-ps**

### CODE REVIEW

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java [hunk 1]
```diff
@@ -799,4 +824,32 @@ private static JsonResponseContactPosition toJson(@Nullable final Job job)
...
+		return JsonProductBPartner.builder()
+				.bpartnerId(JsonMetasfreshId.of(record.getC_BPartner_ID()))
+				.productId(JsonMetasfreshId.of(record.getM_Product_ID()))
+				//
+				.productNo(record.getProductNo())
+				.productName(record.getProductName())
+				.productDescription(record.getProductDescription())
+				.productCategory(record.getProductCategory())
+				//
+				.ean(record.getUPC())
```
Comment (metas-ts) [inline backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java:840]:
```suggestion
				.ean(record.getEAN_CU())
```


..or do i miss sth?

File: backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java [hunk 2]
```diff
@@ -126,31 +127,34 @@ public void updateOrgIdByRecordIdAndType(
...
-		final ImmutableMap.Builder<ExternalReferenceQuery, ExternalReference> result = ImmutableMap.builder();
-		for (final ExternalReferenceQuery query : queries)
-		{
-			final ExternalReference externalReference = query2reference.get(query);
-			result.put(query, CoalesceUtil.coalesce(externalReference, ExternalReference.NULL));
-		}
+		final Map<ExternalReferenceQuery, ExternalReference> result = new HashMap<>();
 
-		return result.build();
+		externalReferences.forEach(externalReference -> queries.forEach(query -> {
```
Comment (metas-ts) [inline backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java:148]: this is above my paygrade 😂 
please convert to old-school for-loops

File: backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java [hunk 3]
```diff
@@ -126,31 +127,34 @@ public void updateOrgIdByRecordIdAndType(
...
+			if (query.matches(externalReference))
+			{
+				result.put(query, externalReference);
+			}
+		}));
+
+		queries.forEach(query -> result.putIfAbsent(query, ExternalReference.NULL));
```
Comment (metas-ts) [inline backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java:155]: (ok, here i guess it's fine)

File: backend/de.metas.externalreference/src/main/java/de/metas/externalreference/rest/v1/ExternalReferenceRestController.java [hunk 4]
```diff
@@ -43,8 +43,7 @@
 
 @RequestMapping(value = {
 		MetasfreshRestAPIConstants.ENDPOINT_API_DEPRECATED + "/externalRef",
-		MetasfreshRestAPIConstants.ENDPOINT_API_V1 + "/externalRef",
-		MetasfreshRestAPIConstants.ENDPOINT_API_V2 + "/externalRefs" })
+		MetasfreshRestAPIConstants.ENDPOINT_API_V1 + "/externalRef" })
 @RestController
 @Profile(Profiles.PROFILE_App)
```
Comment (metas-ts) [inline backend/de.metas.externalreference/src/main/java/de/metas/externalreference/rest/v1/ExternalReferenceRestController.java:48]:
```suggestion
@Profile(Profiles.PROFILE_App)
@Deprecated
```

(let's flag everything `v1` as deprecated, at least when we touch it)

**Review by metas-ts — CHANGES_REQUESTED** [review]
please check the few remarks I had and merge `tasty_ham_uat`into your branch to fix the conflicts. other than that lgtm

File: backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java (hunk 2 above)
Comment (dm-ps) [reply backend/de.metas.externalreference/src/main/java/de/metas/externalreference/ExternalReferenceRepository.java]: Converted 😄

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java (hunk 1 above)
Comment (dm-ps) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java:840]: There's a similar method in GetProductsCommand: de.metas.rest_api.v2.product.command.GetProductsCommand#toJsonProductBPartner
I've inspired from there, that's why I used UPC, but I can change it to EAN_CU if needed

Comment (dm-ps) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java:840]: Also, from the current description seems to make sense to use it as `ean`

image

Comment (metas-ts) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java:840]: i see the point, but that evidence is misleading..thx for the hint regarding GetProductsCommand..i'm fixing it there as well

Comment (metas-ts) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java:840]: ah, no..there in M_Product we don't have EAN..ther it's kindof right (the column naming is unfortunate though..)

**Review by metas-ts — APPROVED** [review]

**Pull request merged**